front/pages/i/_shortId/_slug.vue

```diff
@@ -177,28 +177,32 @@ export default {
   $_veeValidate: {
     validator: 'new'
   },
-  async asyncData({ app, route, store, error }) {
-    const isMyIdea = store.getters['userData/userId'] === route.params.userId
+
+  async asyncData({ app, route, store, error, res }) {
     try {
       const { data } = await app.$amplifyApi.graphql({
-        query: isMyIdea ? getMyIdea : getUsersIdea,
-        variables: {
-          userId: route.params.userId,
-          ideaId: route.params.ideaId
-        },
-        authMode: isMyIdea ? undefined : 'API_KEY'
+        query: getIdea,
+        variables: { shortId: route.params.shortId },
+        authMode: 'API_KEY'
       })
-      return {
-        idea: data[isMyIdea ? 'getMyIdea' : 'getUsersIdea'],
-        isMyIdea
+      const idea = data.getIdea
+
+      // Redirect to correct slug if it doesn't match
+      if (idea.slug !== route.params.slug) {
+        res.writeHead(301, { Location: `/i/${idea.shortId}/${idea.slug}` })
```

Comment:
thats not correct. res will onky exist on ssr, while this can be executed on client. use redirect method from context

Reply:
What is the correct way of 301 redirect then? This is the only way I googled. 
I can check it it's on SSR and run different redirect for client.

> Also different field might be for user own idea vs not own

Now sure what do you mean here?

Reply:
Actually looks like I found this in docs, pushed changes.